listener/listener.go

```diff
@@ -260,37 +275,23 @@ func (l *Listener) Listen(ctx context.Context, handler message.Handler, topicNam
 	}()
 
 	// Generate new subscription client
+	// TODO: Add servicebus.SubscriptionWithPrefetchCount(prefetch count) option and expose the setting via the listener
 	sub, err := topic.NewSubscription(l.subscriptionEntity.Name)
 	if err != nil {
 		return fmt.Errorf("failed to create new subscription %s: %w", l.subscriptionEntity.Name, err)
 	}
-	subReceiver, err := sub.NewReceiver(ctx)
+
+	var receiverOpts []servicebus.ReceiverOption
+	if l.PrefetchCount != nil {
+		receiverOpts = append(receiverOpts, servicebus.ReceiverWithPrefetchCount(*l.PrefetchCount))
```

Comment:
I'm assuming the TODO on line 278 might be for this, but I wanted to let you know that this setting can be applied if the subscription has a prefetch count (Also, could be overridden if there was one as the following option is applied after; however, I don't see how the subscription would be non-nil at that point):
https://github.com/Azure/azure-service-bus-go/blob/12810d63bcdc551ef6446e18e08815cdbb2fc74d/subscription.go#L143

Reply:
ah, so azure servicebus exposes it on both Subscription and NewReceiver.
setting it on the Subscription object ends up passing it down to the receiver. I pass it directly to the receiver: 

```golang
// NewReceiver will create a new Receiver for receiving messages off of the queue
func (s *Subscription) NewReceiver(ctx context.Context, opts ...ReceiverOption) (*Receiver, error) {
	ctx, span := s.startSpanFromContext(ctx, "sb.Subscription.NewReceiver")
	defer span.End()

	opts = append(opts, ReceiverWithReceiveMode(s.receiveMode))

	if s.prefetchCount != nil {
		opts = append(opts, ReceiverWithPrefetchCount(*s.prefetchCount))
	}

	return s.namespace.NewReceiver(ctx, s.Topic.Name+"/Subscriptions/"+s.Name, opts...)
}
```